Update Terraform Install Docs (#1041)
harness/developer-hub: docs/continuous-delivery/cd-advanced/terraform-category/terraform-how-tos.md

See [Build custom delegate images with third-party tools](/docs/platform/2_Delegates/customize-delegates/build-custom-delegate-images-with-third-party-tools.md).

> ### Target OS and Architecture

Reviewer: ```suggestion
### Target OS and architecture
```

Reviewer: we use sentence case for headings: https://harness.atlassian.net/wiki/spaces/DOCS/pages/21130772574/Documentation+style+guide

You will need to change the following scripts based on the operating system and architecture of the machine that is running the delegate.

> Linux on x86(Intel or AMD): linux_amd64
> Linux on Arm: linux_arm64
> MacOs on x86(Intel): darwin_amd64
> MacOs on ARM(Apple): darwin_arm64

Reviewer: ```suggestion
- Linux on x86(Intel or AMD): linux_amd64
- Linux on Arm: linux_arm64
- MacOs on x86(Intel): darwin_amd64
- MacOs on ARM(Apple): darwin_arm64
```

### Custom Delegate Image

The reccomended way to install third party tools on your delegate is to create your own delegate image, push it to a container registry, and then to modify your delegate deployments to use your new custom image.

```dockerfile
ARG DELEGATE_TAG=23.03.78705
ARG DELEGATE_IMAGE=harness/delegate
FROM $DELEGATE_IMAGE:$DELEGATE_TAG

USER 0

RUN useradd -u 1001 -g 0 harness

# Install TF
RUN curl -LO  https://releases.hashicorp.com/terraform/1.3.1/terraform_1.3.1_linux_amd64.zip \
  && unzip -q terraform_1.3.1_linux_amd64.zip \
  && mv ./terraform /usr/bin/ \
  && terraform --version

USER 1001
```

> ### Init Script
> 
> Alternativly you can install terraform when a delegate pod is launched by specifying an initilization script in the delegates environment.

Reviewer: ```suggestion
### Init script

Alternatively, you can install Terraform when a delegate pod is launched by specifying an initialization script in the delegate YAML `INIT_SCRIPT` [environment variable](https://developer.harness.io/docs/platform/delegates/delegate-reference/delegate-environment-variables/).
```
